rules/TypeDeclaration/ValueObject/AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclaration.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\ValueObject;
+
+use PHPStan\Type\ObjectType;
+use Rector\Validation\RectorAssert;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddParamTypeForFunctionLikeWithinCallLikeDeclarationRector\AddParamTypeForFunctionLikeWithinCallLikeDeclarationRectorTest
+ */
+final readonly class AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclaration
+{
+    /**
+     * @param int<0, max>|string $callLikePosition
+     * @param int<0, max> $functionLikePosition
+     * @param int<0, max>|string $fromArgPosition
+     */
+    public function __construct(
+        private string $className,
+        private string $methodName,
+        private int|string $callLikePosition,
```

Comment:
Support for `int` is enough here

Reply:
Are you sure? If it's an arg it could be a named one and that might cause an issue for some people. It makes little difference to the rule but make rules more compatible with code.